baseapp/baseapp.go

```diff
@@ -534,53 +538,74 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		ctx = ctx.WithMultiStore(msCache)
 	}
 
-	finalResult := sdk.Result{}
-	var logs []string
-	for i, msg := range msgs {
+	// accumulate results
+	logs := make([]string, 0, len(msgs))
+	var data []byte   // NOTE: we just append them all (?!)
```

Comment:
I'm still not clear what exactly `data` and `logs` are for, and what the difference between them is intended to be. 

At the moment, I think the SDK uses `data` in one instance to return a proposal ID (in governance), and doesn't use logs at all. The created proposal ID could just as easily be returned as a tag (probably will be anyways, for later querying) - maybe we can simplify ABCI or what parts of ABCI we use?

Reply:
`log` is intended for non-deterministic output from the app. There is some sense that it would be used to pipe the apps actual logs back to tendermint but I don't think anyone has ever done that.

`data` is a deterministic result - we don't really use it AFAIK, so it probably could just be absorbed into tags, but it might be worth keeping for simple apps that want a definitive return value